ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java

```diff
@@ -240,20 +233,42 @@ private boolean addTransaction(final TransactionInfo transactionInfo) {
   }
 
   private boolean addTransactionForSenderAndNonce(final TransactionInfo transactionInfo) {
-    final Map<Long, TransactionInfo> transactionsForSender =
-        transactionsBySender.computeIfAbsent(transactionInfo.getSender(), key -> new TreeMap<>());
     final TransactionInfo existingTransaction =
-        transactionsForSender.get(transactionInfo.getNonce());
+        getTrackedTransactionBySenderAndNonce(transactionInfo);
     if (existingTransaction != null) {
       if (!shouldReplace(existingTransaction, transactionInfo)) {
         return false;
       }
       removeTransaction(existingTransaction.getTransaction());
     }
-    transactionsForSender.put(transactionInfo.getNonce(), transactionInfo);
```

Comment:
The bug was here where we would try updating this local variable `transactionsForSender` after it [had been removed](https://github.com/hyperledger/besu/pull/93/files#diff-408daf8f75a64fdf939fdbc024777b10L160) from the instance variable `transactionsBySender`.  In this case, the new transaction would not be tracked so that subsequent duplicate transactions would fail to eject this transaction.